preppal-fe/src/components/nav-bar/nav-bar.tsx

```diff
@@ -2,10 +2,28 @@ import Nav from 'react-bootstrap/Nav';
 import Navbar from 'react-bootstrap/Navbar';
 import { Form } from 'react-bootstrap';
 import './nav-bar.css'
-import { useState } from 'react';
+import React from 'react';
 
 const NavBar = () => {
-  const [loggedIn] = useState(false)
+  const [loggedIn, setLoggedIn] = React.useState(false)
+  const [user, setUser] = React.useState('')
+
+  React.useEffect(() => {
+    fillUserContent();
+    // eslint-disable-next-line
+  }, []);
+
+  const fillUserContent = async () => {
+    const token = localStorage.getItem("token");
+    try {
+      if (token !== "undefined")
```

Comment:
bad boolean, this should be if (token)

Reply:
Just to make the code easier to read I wrote if(token !== null) which would mean the same check.

Reply:
I believe the check for undefined and null are separate. So in this case when you check if it's not null and token is undefined, the boolean value will be true. There is a way to more loosely check so that null and undefined are both covered but just checking if the token is a truthy value is better

Reply:
Thank you for telling me this, I didn't even know that truthy was something that existed in this language until now. It has been updated accordingly.